src/azure-cli-core/azure/cli/core/commands/arm.py

```diff
@@ -312,6 +324,106 @@ def __call__(self, parser, namespace, values, option_string=None):
     main_command_table[name] = cmd
     main_command_module_map[name] = module_name
 
+def cli_generic_wait_command(module_name, name, getter_op, factory=None, table_transformer=None):
+
+    if not isinstance(getter_op, string_types):
+        raise ValueError("Getter operation must be a string. Got '{}'".format(getter_op))
+    get_arguments_loader = lambda: dict(extract_args_from_signature(get_op_handler(getter_op)))
+
+    def arguments_loader():
+        arguments = {}
+        arguments.update(get_arguments_loader())
+        return arguments
+
+    def get_provisioning_state(instance):
+        provisioning_state = getattr(instance, 'provisioning_state', None)
+        if not provisioning_state:
+            #some SDK, like resource-group, has 'provisioning_state' under 'properties'
+            properties = getattr(instance, 'properties', None)
+            if properties:
+                provisioning_state = getattr(properties, 'provisioning_state', None)
+        return provisioning_state
+
+    def handler(args):
+        from msrest.exceptions import ClientException
+        import time
+        try:
+            client = factory() if factory else None
+        except TypeError:
+            client = factory(None) if factory else None
+
+        getterargs = {key: val for key, val in args.items()
+                      if key in get_arguments_loader()}
+
+        getter = get_op_handler(getter_op)
+
+        timeout = args.pop('timeout')
+        interval = args.pop('interval')
+        wait_for_created = args.pop('created')
+        wait_for_deleted = args.pop('deleted')
+        wait_for_updated = args.pop('updated')
+        wait_for_exists = args.pop('exists')
+        custom_condition = args.pop('custom')
+        if not any([wait_for_created, wait_for_updated, wait_for_deleted,
+                    wait_for_exists, custom_condition]):
+            raise CLIError("Please specify at least one flag from 'create', 'updated', 'delete', 'exits', or 'property'")#pylint: disable=line-too-long
```

Comment:
1) Looks like 'property' was renamed 'custom' so this just needs to be updated. Also, 'exists' is misspelled 'exits'.

2) We might consider having a default here instead of raising an error when nothing is specified.

3) Also, recommend this form for the error: "incorrect usage: --created | --updated | --deleted | --exists | --custom JMESPATH"